restcomm/restcomm.core/src/main/java/org/restcomm/connect/core/service/RestcommConnectServiceProvider.java

```diff
@@ -96,4 +101,15 @@ public ProfileService provideProfileService() {
      */
     public RecordingService recordingService() { return recordingService; }
 
+    /**
+     *
+     * @return
+     */
+    public UriUtils uriUtils() { return uriUtils; }
+
+    //Used for unit testing - not elegant way though
```

Comment:
agree on not being elegant, we should create story about this tech debt on serviceProvider not being testable so we can tackle it in tech debt sprint

Reply:
@maria-farooq @hrosa This patch and needed to mock UriUtils for VoiceInterpreter unit tests:
- org.restcomm.connect.interpreter.mediagroup.StopMediaGroupTest
- org.restcomm.connect.interpreter.GatherSpeechTest

For those unit tests ServletContext is not avalilable and in my attempt to mock servlet context I hit a wall on the fact that org.restcomm.connect.core.service.RestcommConnectServiceProvider is a singleton and cannot be mocked by Mockito. I even tried with PowerMock with no success. 

It would be nice if you can provide a concrete solution to avoid this ugly patch.